### PR TITLE
UA_StatusCodes equality checks fail if additional information bits are present inside open62541 lib

### PR DESCRIPTION
**Pull request by rtorresmatheus**
Changes:

- Added condition guard for OPC UA variable;
- Added new version of open52641 lib to fix StatusCode. See #[5376](https://github.com/open62541/open62541/issues/5376).

### CODE REVIEW

**Review by AntonioJanael — APPROVED** [review]
Great job!

File: src/iface-opc-ua.c [hunk 1]
```diff
@@ -240,21 +260,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
 	rc = setup_read_data(value, data_item);
 	memset(&tmp, 0, sizeof(tmp));
 
-	if (rc == UA_STATUSCODE_GOOD) {
-		rc = get_read_value(value, data_item->schema.value_type,
+	if (!((rc & UA_STATUSCODE_BAD) || (rc & UA_STATUSCODE_UNCERTAIN))) {
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:263]: I would recommend replacing the 'rc' variable with a name with more details.

File: src/iface-opc-ua.c [hunk 2]
```diff
@@ -240,21 +260,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
...
+		get_read_value(value, data_item->schema.value_type,
 				    data_item->value_type_size, &tmp);
 	}
 
-	if (rc & UA_STATUSCODE_BAD) {
+	if (rc & UA_STATUSCODE_BADSERVERNOTCONNECTED) {
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:268]: I would recommend replacing the 'rc' variable with a name with more details.

File: src/iface-opc-ua.c [hunk 3]
```diff
@@ -240,21 +260,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
...
 		l_error("Failed to read from OPC UA: %#010x",
 			rc);
 		l_io_destroy(opc_ua_io);
 		opc_ua_io = NULL;
 		rental = -EINTR;
-	} else if (rc & UA_STATUSCODE_UNCERTAIN) {
+	} else if (((rc & UA_STATUSCODE_BAD) ||
+		   (rc & UA_STATUSCODE_UNCERTAIN))) {
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:275]: I would recommend replacing the 'rc' variable with a name with more details.

File: src/iface-opc-ua.c [hunk 4]
```diff
@@ -240,21 +260,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
...
 		l_error("Failed to read from OPC UA for Data_%d: %#010x",
 			data_item->sensor_id, rc);
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:277]: I would recommend replacing the 'rc' variable with a name with more details.

File: src/iface-opc-ua.c [hunk 5]
```diff
@@ -240,21 +260,22 @@ int iface_opc_ua_read_data(struct knot_data_item *data_item)
 	rc = setup_read_data(value, data_item);
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:260]: I would recommend replacing the 'rc' variable with a name with more details.

File: src/iface-opc-ua.c [hunk 6]
```diff
@@ -134,94 +134,114 @@ static UA_StatusCode setup_read_data(UA_Variant *value,
 			 data_item->identifier);
 	rc = UA_Client_readValueAttribute(opc_ua_client,
 						  *parent, value);
-
 	free(parent);
 	return rc;
```
Comment (luisfelipemisil) [inline src/iface-opc-ua.c:138]: I would recommend replacing the 'rc' variable with a name with more details.

**Review by luisfelipemisil — APPROVED** [review]
Great job!

**Pull request merged**